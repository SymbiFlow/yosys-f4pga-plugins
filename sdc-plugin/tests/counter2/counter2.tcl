yosys -import
plugin -i sdc
# Import the commands from the plugins to the tcl interpreter
yosys -import

read_verilog $::env(DESIGN_TOP).v
read_verilog -specify -lib -D_EXPLICIT_CARRY +/xilinx/cells_sim.v
read_verilog -lib +/xilinx/cells_xtra.v
hierarchy -check -auto-top
# Start flow after library reading
synth_xilinx -vpr -flatten -abc9 -nosrl -nodsp -iopad -run prepare:check

# Read the design's timing constraints
read_sdc $::env(DESIGN_TOP).input.sdc

# Write the clocks to file
set fh [open $::env(DESIGN_TOP).txt w]
set clocks [get_clocks]
puts $fh $clocks
close $fh

# Write out the SDC file after the clock propagation step
write_sdc $::env(DESIGN_TOP).sdc
