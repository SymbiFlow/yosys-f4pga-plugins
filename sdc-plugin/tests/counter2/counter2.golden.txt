clk_int_1 clk clk2
